modules/RTC/RTCUtils.js

```diff
@@ -460,14 +473,46 @@ function newGetConstraints(um = [], options = {}) {
         }
 
         constraints.video = {
-            mandatory: {
-                chromeMediaSource: 'desktop',
-                chromeMediaSourceId: options.desktopStream,
-                maxWidth: window.screen.width,
-                maxHeight: window.screen.height,
-                maxFrameRate: 3
-            }
+            mandatory: getSSConstraints({
+                ...options,
+                source: 'desktop'
+            })
+        };
+    }
+
+    return constraints;
+}
+
+/**
+ * Generates GUM constraints for screen sharing.
+ *
+ * @param {Object} options - The options passed to
+ * <tt>obtainAudioAndVideoPermissions</tt>.
+ * @returns {Object} - GUM constraints.
+ *
+ * TODO: Currently only the new GUM flow and Chrome is using the method. We
+ * should make it work for all use cases.
+ */
+function getSSConstraints(options = {}) {
+    const { max, min }
+        = options.frameRate || {
+            min: SS_DEFAULT_FRAME_RATE,
+            max: SS_DEFAULT_FRAME_RATE
         };
+
+    const constraints = {
+        chromeMediaSource: options.source,
+        chromeMediaSourceId: options.desktopStream,
```

Comment:
If source is 'screen' then options.desktopStream will be undefined. I want to verify with you that chrome doesn't complain about it being undefined.